test/lib/epochs.test.js

```diff
@@ -313,14 +310,15 @@ test('lib/epochs (getPreferredEpoch - 4.4. same membership)', async (t) => {
     ])
   )
 
-  let livePreferredEpoch = {}
+  const livePreferredEpochs = []
+  let testsRunning = true
   pull(
     Epochs(oscar).getPreferredEpoch.stream(group.id, { live: true }),
     pull.drain(
-      (epoch) => {
-        livePreferredEpoch = epoch
-      },
-      (err) => {}
+      (epoch) => livePreferredEpochs.unshift(epoch),
+      (err) => {
+        if (err && testsRunning) t.error(err, 'getPreferredEpoch.stream smooth')
```

Comment:
because there was no error handling on the drain, errors were getting swallowed :cry: which wasted a lot of time.
This is an anti-pattern of drain IMO - I think it would be smarter to require an error handling function even if it's `() => {}`

Reply:
well there was one but it was `(err)=> {}`